db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2256,19 +2269,36 @@ ColumnFamilyData* DBImpl::PickCompactionFromQueue(
 
 void DBImpl::SchedulePendingFlush(const FlushRequest& flush_req,
                                   FlushReason flush_reason) {
+  mutex_.AssertHeld();
   if (flush_req.empty()) {
     return;
   }
-  for (auto& iter : flush_req) {
-    ColumnFamilyData* cfd = iter.first;
-    cfd->Ref();
-    cfd->SetFlushReason(flush_reason);
+  if (!immutable_db_options_.atomic_flush) {
+    // For the non-atomic flush case, we never schedule multiple column
```

Comment:
Is there a need to impose this restriction here? It seems like the whole code flow could be simpler if we simply loop through ```flush_req``` here and queue a column family if its not been queued before, skip it otherwise. Maybe I'm missing something.

Reply:
In the non-atomic flush case, we would like bg flush for different column families to run in different bg threads. This is actually the behavior before atomic flush was introduced. The advantage is that flush can finish more quickly to avoid write stalls when there are sufficient threads in the pool. This can be beneficial for the case of remote storage, too.